utils/service_discovery/sd_backend.py

```diff
@@ -0,0 +1,218 @@
+                      'will be left unconfigured.' % (c_id, image))
+            return None
+
+        check_name, init_config_tpl, instance_tpl, variables = template_config
+        var_values = {}
+        for v in variables:
+            if v in self.VAR_MAPPING:
+                var_values[v] = self.VAR_MAPPING[v](inspect)
+            else:
+                log.debug("Didn't find any way to extract the value for %s, "
+                          "looking in env variables/docker labels..." % v)
+                var_values[v] = self._get_explicit_variable(inspect, v)
+        init_config, instances = self._render_template(init_config_tpl or {}, instance_tpl or {}, var_values)
+        return (check_name, init_config, instances)
+
+    def _get_template_config(self, image_name):
+        """Extract a template config from a K/V store and returns it as a dict object."""
+        config_backend = self.agentConfig.get('sd_config_backend')
+        if config_backend is None:
+            auto_conf = True
+            log.info('No supported configuration backend was provided, using auto-config only.')
+        else:
+            auto_conf = False
+
+        tpl = self.config_store.get_check_tpl(image_name, auto_conf=auto_conf)
+
+        if tpl is not None and len(tpl) == 3:
+            check_name, init_config_tpl, instance_tpl = tpl
+        else:
+            log.debug('No template was found for image %s, leaving it alone.' % image_name)
+            return None
+        try:
+            # build a list of all variables to replace in the template
+            variables = self.PLACEHOLDER_REGEX.findall(str(init_config_tpl)) + \
+                self.PLACEHOLDER_REGEX.findall(str(instance_tpl))
+            variables = map(lambda x: x.strip('%'), variables)
+            if not isinstance(init_config_tpl, dict):
+                init_config_tpl = json.loads(init_config_tpl)
+                if not isinstance(instance_tpl, dict):
+                    instance_tpl = json.loads(instance_tpl)
+        except json.JSONDecodeError:
+            log.error('Failed to decode the JSON template fetched from {0}.'
+                      'Auto-config for {1} failed.'.format(config_backend, image_name))
+            return None
+        return [check_name, init_config_tpl, instance_tpl, variables]
+
+    def _get_explicit_variable(self, container_inspect, var):
+        """Extract the value of a config variable from env variables or docker labels.
+           Return None if the variable is not found."""
+        conf = self._get_config_space(container_inspect['Config'])
+        if conf is not None:
+            return conf.get(var)
+
+    def _get_config_space(self, container_conf):
+        """Check whether the user config was provided through env variables or container labels.
+           Return this config after removing its `datadog_` prefix."""
+        env_variables = {v.split("=")[0].split("datadog_")[1]: v.split("=")[1]
```

Comment:
We should remove that i think. It's not documented and i guess it's based on the quick POC i made in the early days. I don't think we should keep it